impeller/docs/vulkan_threading.md

```diff
@@ -0,0 +1,61 @@
+# Threading in Vulkan
+
+The Vulkan backend uses a dedicated concurrent worker pool that is created along
+with the creation of the Vulkan context.
+
+Unlike other pools such as the IO worker pool, long running tasks may **not** be
+posted to this pool. This is because frame workloads can and often are
+distributed to workers in this pool. Having a potentially long running task
+(such as texture decompression) may potentially block frame critical tasks. The
+limitation of a separate pool for frame critical tasks is working around the
+separate limitation of not being able to specify a QoS to specific tasks and may
+be lifted in the future.
+
+There is also a separate component called the fence waiter operates on its own
+thread. The sole purpose of this thread is to wait for the GPU to complete
+processing commands so that Impeller can delete the resources when the GPU is
```

Comment:
I would phrase this as something like "The purpose of the fence waiter is to ensure that the resource/ref count lives _at least_ as long as the GPU command buffer(s) that access this resource". 

Because of course these fences can fire quite quickly on smaller workloads and finish before the overall frame is finished. Without the resource manager, this would lead to resource deletion as part of the frame workload.

Reply:
Done. Copied almost verbatim.